SPARCore/GADArticle.m

```diff
@@ -1,7 +1,107 @@
+                                                                           *_Nullable articles,
+                                                            NSError *_Nullable error))completion {
+    NSURL *queryURL = [GADArticle createURLWithPublication:publicationId];
+    
+    [GADRemoteModel fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData) {
+                            return [GADArticle articlesFromJSON:jsonData];
+                         }
+                        completionHandler:completion];
+
+}
+
+- (void) populateArticleWithId: (void(^_Nonnull)(GADRemoteModel * *_Nullable model,
+                                                NSError *_Nullable error))completion {
+    NSURL *queryURL = [GADArticle createURLWithArticle:self.articleId publication:self.publicationId];
+    
+    [GADRemoteModel fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData){
+                             return [GADArticle articlesFromJSON:jsonData];
+                         }
+                        completionHandler:^(NSArray <GADArticle *> *_Nullable model, NSError *_Nullable error){
+                            GADArticle *fullArticle = model[0];
+                            self.content = fullArticle.content;
+                            self.authors = fullArticle.authors;
+                            //completion(); --How should we call the completion handler here?
+                        }];
+    
+}
+
++ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
+    
+    NSMutableArray <GADArticle *> *articles = [[NSMutableArray alloc] init];
+    NSError *error = nil;
+    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
+                                                         options:kNilOptions error:&error];
+    
+    if (error != nil) {
+        NSLog(@"Error parsing JSON.");
+    }
+    
+    for (NSDictionary *element in jsonArray) {
+        //Map fields of element to fields of article
+        GADArticle *article = [[GADArticle alloc] init];
+        //datePublished field is a UNIX Timestamp number - converting to NSDate here
+        int timeStamp = (int)element[API_DATE_PUBLISHED];
+        article.datePublished = [NSDate dateWithTimeIntervalSince1970: timeStamp];
+        article.brief = element[API_BRIEF];
+        article.headerImage = element[API_HEADER_IMAGE];
+        article.publicationId = element[API_PUBLICATION_ID];
+        article.articleId = element[API_ARTICLE_ID];
+        article.title = element[API_TITLE];
+        if (element[API_CONTENT]) {
+            article.content = element[API_CONTENT];
+            article.authors = element[API_AUTHORS];
+        }
+        [articles addObject:article];
+    }
+    return articles;
+}
+
++ (NSURL *) createBaseURL {
```

Comment:
Drop `create`. Just `baseURL`. It's cleaner 

https://www.youtube.com/watch?v=dU6scly2AFU